lib/puppet/defaults.rb

```diff
@@ -32,6 +32,41 @@ def self.valid_file_checksum_types
       %w[sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite md5 md5lite mtime ctime]
   end
 
+  def self.log_ca_migration_warning
+    urge_to_migrate = <<-UTM
+The cadir is currently in your puppet/ssl directory. This location will be
+deprecated and removed in a future version of puppet. Please run the the
+puppetserver ca tool to migrate it out to the puppetserver directory. Use
+`puppetserver ca migrate --help` for more info.
+UTM
+    Puppet.warn_once('deprecations',
+                     'CA migration message',
+                     urge_to_migrate,
+                     :default,
+                     :default)
+  end
+
+  def self.default_cadir
+    old_ca_dir = "#{Puppet[:ssldir]}/ca"
+    new_ca_dir = "/etc/puppetlabs/puppetserver/ca"
+
+    # If the file is a symlink, then we have run the migration tool
+    if File.exist?(old_ca_dir)
+      if !File.symlink?(old_ca_dir)
+        log_ca_migration_warning
+        old_ca_dir
+      else
+        target = File.readlink(old_ca_dir)
+        if target.start_with?(Puppet[:ssldir])
+          log_ca_migration_warning
+        end
+        target
+      end
```

Comment:
(nit) Can we maybe switch these around so that we're not doing `if !`? I feel like that is clearer to read.

Reply:
Yeah, that sounds reasonable.